ipywidgets/widgets/widget.py

```diff
@@ -223,6 +272,10 @@ def open(self):
             if buffers:
                 # FIXME: workaround ipykernel missing binary message support in open-on-init
                 # send state with binary elements as second message
+                # TODO: if this gets fixed, _split_state_buffers does not need to have a seperate
```

Comment:
Basically, we think of the binary serialization as just a detail of the actual wire format (which it is), so we keep those details as close to the wire as we can.